src/test/java/eu/yals/test/ui/VaadinTest.java

```diff
@@ -30,155 +30,165 @@
-
-      browsers.add(chrome);
+        sauceLabsLog.setLevel(Level.SEVERE);
+        selenideRemoteLog.setLevel(Level.WARNING);
     }
 
-    if (testBrowsers.contains(TestApp.Browser.FIREFOX)) {
-      browsers.add(BrowserUtil.firefox());
+    private static final int SERVER_PORT =
+            Integer.parseInt(
+                    System.getProperty(TestApp.Properties.SERVER_PORT, TestApp.Defaults.SERVER_PORT));
+    private static final String LOCAL_URL =
+            String.format("http://host.testcontainers.internal:%d", SERVER_PORT);
+    private static final String REPORT_DIRECTORY =
+            System.getProperty(TestApp.Properties.REPORT_DIR, TestApp.Defaults.REPORT_DIR);
+    protected static final String BASE_URL =
+            System.getProperty(TestApp.Properties.TEST_URL, LOCAL_URL);
+
+    private static final String BUILD_NAME =
+            System.getProperty(TestApp.Properties.BUILD_NAME, TestApp.Defaults.BUILD_NAME);
+
+    private static String testName;
+
+    @Rule
+    public final TestRule watchman =
+            new TestWatcher() {
+                @Override
+                protected void starting(final Description description) {
+                    super.starting(description);
+                    testName = setTestNameFromTestDescription(description);
+                    log.info(String.format("Starting build '%s'. Test: '%s", BUILD_NAME, testName));
+                }
+
+                @Override
+                protected void succeeded(final Description description) {
+                    super.succeeded(description);
+                    Cookie cookie = new Cookie("zaleniumTestPassed", "true");
+                    getDriver().manage().addCookie(cookie);
+                }
+
+                @Override
+                protected void failed(final Throwable e, final Description description) {
+                    Cookie cookie = new Cookie("zaleniumTestPassed", "false");
+                    getDriver().manage().addCookie(cookie);
+                }
+            };
+
+    /**
+     * Performs actions needed for running tests.
+     *
+     * @throws Exception if error occurred
+     */
+    @Before
+    public void setup() throws Exception {
+        addTestNameToDriver();
+
+        super.setup();
+        Parameters.setScreenshotErrorDirectory(REPORT_DIRECTORY);
     }
-    if (testBrowsers.contains(TestApp.Browser.SAFARI)) {
-      browsers.add(BrowserUtil.safari());
+
+    protected void open(final String relativeOrAbsoluteUrl) {
+        final String PROTOCOL_MARKER = "://";
+        boolean isUrlAbsolute = relativeOrAbsoluteUrl.contains(PROTOCOL_MARKER);
+        if (isUrlAbsolute) {
+            getDriver().get(relativeOrAbsoluteUrl);
+        } else {
+            getDriver().get(BASE_URL + relativeOrAbsoluteUrl);
+        }
     }
-    if (testBrowsers.contains(TestApp.Browser.IE)) {
-      browsers.add(BrowserUtil.ie11());
+
+    protected YalsElement $$(final String cssSelector) {
```

Comment:
![Codacy](https://app.codacy.com/assets/images/favicon.png) Issue found: [The instance method name '$$' doesn't match '[a-z][a-zA-Z0-9]*'](https://app.codacy.com/gh/yadevee/yals/pullRequest?prid=5409395)